tests/functional/spec/list_collector_section_summary_items.spec.js

```diff
@@ -0,0 +1,190 @@
+      anyMoreCompaniesYes();
+      expect($(AnyCompaniesOrBranchesAddPage.companyOrBranchName()).isExisting()).to.be.true;
+      expect($(AnyCompaniesOrBranchesAddPage.registrationNumber()).isExisting()).to.be.true;
+      expect($(AnyCompaniesOrBranchesAddPage.authorisedInsurerRadioYes()).isExisting()).to.be.true;
+      expect($(AnyCompaniesOrBranchesAddPage.heading()).getText()).to.contain(
+        "Give details about the company or branch that undertakes general insurance business"
+      );
+    });
+    it("When I add my own item and relevant data, Then I should be able to edit that item from the section summary page.", () => {
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      answerUkBasedQuestion();
+      $(SectionSummaryPage.companiesListEditLink(1)).click();
+      expect(browser.getUrl()).to.contain("edit-company/?return_to=section-summary");
+    });
+    it("When I decide not to add an item and relevant data and I change my answer to yes, Then I should be able to add the item.", () => {
+      drivingQuestionNo();
+      answerUkBasedQuestion();
+      expect(browser.getUrl()).to.contain(SectionSummaryPage.url());
+      expect($(SectionSummaryPage.companiesListEditLink(1)).isExisting()).to.be.false;
+      expect($(SectionSummaryPage.companiesListRemoveLink(1)).isExisting()).to.be.false;
+      expect($(SectionSummaryPage.companiesListAddLink()).isExisting()).to.be.false;
+      $(SectionSummaryPage.anyCompaniesOrBranchesAnswerEdit()).click();
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      expect(browser.getUrl()).to.contain(SectionSummaryPage.url());
+      expect($(SectionSummaryPage.companiesListEditLink(1)).isExisting()).to.be.true;
+      expect($(SectionSummaryPage.companiesListRemoveLink(1)).isExisting()).to.be.true;
+      expect($(SectionSummaryPage.companiesListAddLink()).isExisting()).to.be.true;
+    });
+    it("When I decide to add an item and relevant data and I decide to remove it, Then I should be able to see the item again after I decide to add more items.", () => {
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      answerUkBasedQuestion();
+      $(SectionSummaryPage.anyCompaniesOrBranchesAnswerEdit()).click();
+      drivingQuestionNo();
+      expect(browser.getUrl()).to.contain(SectionSummaryPage.url());
+      expect($(SectionSummaryPage.companiesListEditLink(1)).isExisting()).to.be.false;
+      expect($(SectionSummaryPage.companiesListRemoveLink(1)).isExisting()).to.be.false;
+      $(SectionSummaryPage.anyCompaniesOrBranchesAnswerEdit()).click();
+      drivingQuestionYes();
+      expect(browser.getUrl()).to.contain(SectionSummaryPage.url());
+      expect($(SectionSummaryPage.companiesListEditLink(1)).isExisting()).to.be.true;
+      expect($(SectionSummaryPage.companiesListRemoveLink(1)).isExisting()).to.be.true;
```

Comment:
I would check the value, and that placeholder text doesn't exit, no does the add link.

Reply:
Added additional checks 👍